src/main/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnector.java

```
@@ -0,0 +1,250 @@
+import io.cdap.cdap.etl.api.connector.Connector;
+import io.cdap.cdap.etl.api.connector.ConnectorSpec;
+import io.cdap.cdap.etl.api.connector.ConnectorSpecRequest;
+import io.cdap.cdap.etl.api.connector.DirectConnector;
+import io.cdap.cdap.etl.api.connector.SampleRequest;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.plugin.gcp.bigquery.source.BigQuerySourceConfig;
+import io.cdap.plugin.gcp.bigquery.util.BigQueryDataParser;
+import io.cdap.plugin.gcp.common.GCPUtils;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+import javax.annotation.Nullable;
+
+/**
+ * BigQuery Connector Plugin
+ */
+@Plugin(type = Connector.PLUGIN_TYPE)
+@Name(BigQueryConnector.NAME)
+@Description("This connector creates connections to BigQuery, browses BigQuery datasets and tables, sample BigQuery " +
+  "tables. BigQuery is Google's serverless, highly scalable, enterprise data warehouse.")
+public final class BigQueryConnector implements DirectConnector {
+  public static final String NAME = "BigQuery";
+  public static final String ENTITY_TYPE_DATASET = "dataset";
+  public static final String ENTITY_TYPE_TABLE = "table";
+  private static final int ERROR_CODE_NOT_FOUND = 404;
+  private BigQueryConnectorConfig config;
+
+  BigQueryConnector(BigQueryConnectorConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public List<StructuredRecord> sample(SampleRequest sampleRequest) throws IOException {
+    BigQueryPath path = new BigQueryPath(sampleRequest.getPath());
+    String dataset = path.getDataset();
+    String table = path.getTable();
+    if (table == null) {
+      throw new IllegalArgumentException("Path should contain both dataset and table name.");
+    }
+    return getTableData(getBigQuery(), dataset, table, sampleRequest.getLimit());
+  }
+
+  @Override
+  public void test(FailureCollector failureCollector) throws ValidationException {
+    // Should not happen
```

You can remove this if condition, it should fail much earlier when it instantiates this plugin if config is not found

---

done :)